src/app/core/dspace-rest-v2/endpoint-mocking-rest.service.ts

```diff
@@ -0,0 +1,109 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http'
+import { Inject, Injectable } from '@angular/core';
+import { Observable, of as observableOf } from 'rxjs';
+import { GLOBAL_CONFIG, GlobalConfig } from '../../../config';
+import { isEmpty } from '../../shared/empty.util';
+import { RestRequestMethod } from '../data/rest-request-method';
+
+import { DSpaceRESTV2Response } from './dspace-rest-v2-response.model';
+import { DSpaceRESTv2Service, HttpOptions } from './dspace-rest-v2.service';
+import { MOCK_RESPONSE_MAP, MockResponseMap } from './mocks/mock-response-map';
+import * as URL from 'url-parse';
+
+/**
+ * Service to access DSpace's REST API.
+ *
+ * If a URL is found in this.mockResponseMap, it returns the mock response instead
```

Comment:
I'm not sure I understand this service. Could you describe why/when it would be used? If this is only for Specs, we should say that in the description here.

Also, I wonder if this should be moved to the `/mocks` subfolder along with the rest of the related files?

Reply:
Added some documentation on its usage. Also moved it to the `/mock` subfolder.